cpp/include/cudf/io/text/multibyte_split.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/io/text/data_chunk_source.hpp>
+
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <iostream>
+#include <memory>
+
+namespace cudf {
+namespace io {
+namespace text {
+
+std::unique_ptr<cudf::column> multibyte_split(
+  data_chunk_source& source,
```

Comment:
Will the `data_chunk_source` be mutated by this API? Or should it be passed as a `const&`?

Reply:
Right now, no. But there's more to consider if we start caching file reads. Right now `const&` is probably more appropriate.